src/platform/w3dengine/client/w3dbibbuffer.cpp

```diff
@@ -0,0 +1,338 @@
+                    red = 1.0f;
+                }
+
+                if (grn > 1.0f) {
+                    grn = 1.0f;
+                }
+
+                if (blu > 1.0f) {
+                    blu = 1.0f;
+                }
+
+                red = red * 255.0f;
+                grn = grn * 255.0f;
+                blu = blu * 255.0f;
+
+                unsigned int color = (GameMath::Fast_To_Int_Truncate(red) << 16) | (GameMath::Fast_To_Int_Truncate(grn) << 8)
+                    | GameMath::Fast_To_Int_Truncate(blu) | 0xFF000000;
+
+                for (int i = 0; i <= 1; ++i) {
+
+                    if (i == 1) {
+                        m_firstIndex = m_numBibIndices;
+                        m_firstVertex = m_numBibVertices;
+                    }
+
+                    for (int j = 0; j < m_numBibs; ++j) {
+                        if (!m_bibs[j].m_isUnused && m_bibs[j].m_hightlighted == (i != 0)) {
+
+                            int index = m_numBibVertices;
+                            int num_points = 4;
+
+                            if (m_numBibVertices + 6 >= m_maxBibVertex) {
+                                break;
+                            }
+
+                            int num_ind = 6;
+
+                            if (m_numBibIndices + 12 >= m_maxBibIndex) {
+                                break;
+                            }
+
+                            for (int k = 0; k < num_points; ++k) {
+                                float v;
+                                float u;
+
+                                Vector3 point(m_bibs[j].m_points[k]);
+                                switch (k) {
+                                    case 0:
+                                        u = 0.0;
+                                        v = 1.0;
+                                        break;
+                                    case 1:
+                                        u = 1.0;
+                                        v = 1.0;
+                                        break;
+                                    case 2:
+                                        u = 1.0;
+                                        v = 0.0;
+                                        break;
+                                    case 3:
+                                        u = 0.0;
+                                        v = 0.0;
+                                        break;
+                                    default:
+                                        break;
+                                }
+
+                                vertices->u1 = u;
+                                vertices->v1 = v;
+                                vertices->x = point.X;
+                                vertices->y = point.Y;
+                                vertices->z = point.Z;
+                                vertices->diffuse = color;
+
+                                ++vertices;
+                                ++m_numBibVertices;
```

Comment:
I am confused by adding max 4 here, when further above it tests with `m_numBibVertices + 6`

Reply:
dunno why its like that, can confirm its correct to binary